test/integration/auth-methods/oauth2.test.js

```diff
@@ -0,0 +1,260 @@
+var expect = require('chai').expect,
+    sinon = require('sinon');
+
+describe('oauth 2', function () {
+    after(function () {
+        sinon.restore();
+    });
+
+    var testrun,
+        refreshTokenFunction,
+        cancelRefreshTokenFunction,
+        delayFor3000ms = () => {
+            return new Promise((resolve) => {
+                setTimeout(() => {
+                    resolve();
+                }, 3000);
+            });
+        };
```

Comment:
Tip: this can be made generic by accepting a timeout:
```suggestion
        delay = (timeout) => {
            return new Promise((resolve) => {
                setTimeout(() => {
                    resolve();
                }, timeout);
            });
        };
```